### PR TITLE
JDK-8300087: Replace NULL with nullptr in share/cds/

### PR DESCRIPTION
**Pull request by jdksjolen**
Do the conversion in the share/cds/ sub-directory and all of its files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300087](https://bugs.openjdk.org/browse/JDK-8300087): Replace NULL with nullptr in share/cds/

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [de43d4ca](https://git.openjdk.org/jdk/pull/12090/files/de43d4ca14f32a1ba792ee49d874716c01fad077)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to [de43d4ca](https://git.openjdk.org/jdk/pull/12090/files/de43d4ca14f32a1ba792ee49d874716c01fad077)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12090/head:pull/12090` \
`$ git checkout pull/12090`

Update a local copy of the PR: \
`$ git checkout pull/12090` \
`$ git pull https://git.openjdk.org/jdk pull/12090/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12090`

View PR using the GUI difftool: \
`$ git pr show -t 12090`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12090.diff">https://git.openjdk.org/jdk/pull/12090.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/cds/filemap.cpp [hunk 1]
```diff
@@ -2495,7 +2495,7 @@ void FileMapInfo::fixup_mapped_heap_regions() {
 
   // do the same for mapped open archive heap regions
   if (num_open_heap_regions != 0) {
-    assert(open_heap_regions != NULL, "NULL open_heap_regions array with non-zero count");
+    assert(open_heap_regions != nullptr, "null open_heap_regions array with non-zero count");
```
Comment (dholmes-ora) [inline src/hotspot/share/cds/filemap.cpp:2498]: Nit: suggest "Null" to match similar message at L2514.

**Review by dholmes-ora — APPROVED** [review]
LGTM! Thanks.

One suggestion below.

**Review by iklam — APPROVED** [review]
LGTM

**Comment by jdksjolen** [comment]
Passes testing.

/integrate

**Pull request closed without merging**